src/ngraph/op/broadcast.cpp

```diff
@@ -310,6 +322,73 @@ void op::v0::Broadcast::generate_adjoints(autodiff::Adjoints& adjoints, const Ou
     adjoints.add_delta(x, make_shared<op::Sum>(delta, m_broadcast_axes));
 }
 
+namespace
+{
+    template <element::Type_t ET>
+    inline bool
+        evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out, const AxisSet& broadcast_axes)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        runtime::reference::broadcast<T>((arg0->get_data_ptr<ET>()),
+                                         (out->get_data_ptr<ET>()),
+                                         arg0->get_shape(),
+                                         out->get_shape(),
+                                         broadcast_axes);
+        return true;
+    }
+
+    bool evaluate_broadcast(const HostTensorPtr& arg0,
+                            const HostTensorPtr& out,
+                            const AxisSet broadcast_axes,
+                            const Shape output_shape)
+    {
+        if (arg0->get_element_type() != out->get_element_type())
+        {
+            return false;
+        }
```

Comment:
Is this a thing? I would think that would be an invalid broadcast since it is doing a Convert as well.
If this is a valid check then it should throw an exception, just returning false seems wrong.

Reply:
Just set the output element type to the arg0 element type and it will make sure they are compatible. The output element type from validation could be "dynamic" which would not be the same as the arg0 type, but is compatible so setting the element type on output would work.

Reply:
No, i think this should never happen, because the constructor sets node output type and we need to set the HostTensor type.
I will remove this check.